kubectl-cloudflow/version/version.go

```diff
@@ -30,21 +33,58 @@ const ProtocolVersionKey = "protocol-version"
 // ProtocolVersionConfigMapName is the name of the configmap that contains the protocol-version
 const ProtocolVersionConfigMapName = "cloudflow-protocol-version"
 
-// FailOnProtocolVersionMismatch fails and exits if the protocol version of kubectl-cloudflow does not match with the cloudflow operator protocol version.
-func FailOnProtocolVersionMismatch() {
+// CloudflowNamespaceLabelKey is the key of the label for the cloudflow installation namespace
+const cloudflowNamespaceLabelKey = "namespace"
+
+// CloudflowNamespaceLabelValue is the value of the label for the cloudflow installation namespace
+const cloudflowNamespaceLabelValue = "cloudflow-namespace"
+
+// GetProtocolVersionConfigMap Get the protocol version config map set by the operator
+func GetProtocolVersionConfigMap() (*corev1.ConfigMap, error) {
 	k8sClient, k8sErr := k8s.GetClient()
 	if k8sErr != nil {
 		util.LogAndExit("Failed to create new kubernetes client, %s", k8sErr.Error())
 	}
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{cloudflowNamespaceLabelKey: cloudflowNamespaceLabelValue}}
 
 	var cm *corev1.ConfigMap
-	if configMaps, err := k8sClient.CoreV1().ConfigMaps("").List(metav1.ListOptions{}); err == nil {
-		for i, v := range configMaps.Items {
-			if v.Name == ProtocolVersionConfigMapName {
-				cm = &configMaps.Items[i]
-			}
+	configMaps, err := k8sClient.CoreV1().ConfigMaps("").List(
+		metav1.ListOptions{LabelSelector: labels.Set(labelSelector.MatchLabels).String()})
+	if err == nil {
+		if len(configMaps.Items) > 1 {
+			return nil, errors.New("Multiple cloudflow operators detected in namespace .. exiting")
+		}
+		cm = &configMaps.Items[0]
+		if cm.GetName() == ProtocolVersionConfigMapName {
```

Comment:
Now that we check for label and we check that there is only one. why do e need to check for the name as well? I do not think this is very useful and it introduces a corner case and an extra error message that we just don't need.

I suggest we remove this.

Reply:
This is related to the earlier comment that u made. I replied there as well. This is indeed a corner case that can be introduced if we change the name in cli and forget to change in the operator code (or vice versa) .. Should I remove this check ?

Reply:
I think now that we find the configMap by label the check on name has no value anymore.

See above for my reply to your other comment, e.g. it is still possible for the CLI and the operator to have a mismatch in the hardcoded label key/value used to identify the configMap by label. I that is the case we will not find _any_ ConfigMap from the CLI, which is a corner case we should indeed cover and report on.

Reply:
see my earlier reply - will remove that `return` statement ..

Reply:
@debasishg I think we can remove this conditional and just always return here instead of only when the name of the config map is the one we expect. We don't care about the name since we have already found it through its label.

Reply:
done ..